### ISSUE TITLE
Add scene metadata field for original date (of production)

### ISSUE THREAD

**Issue by echo6ix** [stashapp/stash#1078]
# Edit
(2026-07-21): Since writing this, I've learned that my understanding of `production date` was not completely correct. Because the post hinges on that definition, I'm not sure my original conclusions below are as valid as written. But I'll leave the original portion up for context.

### Definition of production date

It is my layman's understanding that under [18 U.S. Code § 2257](https://anonymiz.com/?https%3A%2F%2Fwww.law.cornell.edu%2Fuscode%2Ftext%2F18%2F2257), the legal definition of *produce* is broad. It encompasses not just filming, but explicitly includes "duplicating, reproducing, or reissuing" content.

There are two distinct categories of *produce*:

1. Primary producer, "any person who actually films, videotapes, photographs, or creates a digitally- or computer-manipulated image, a digital image, or a picture of, or who digitizes an image of, a visual depiction of [...]" aka _original date of production_.
2. Secondary producer, "any person who produces, assembles, manufactures, publishes, duplicates, reproduces, or reissues a book, magazine, periodical, film, videotape, or digitally- or computer-manipulated image, picture, or other matter intended for commercial distribution that contains a visual depiction of [...]"

### Why does this matter?

Because both are legally *productions*, and publishers cite the the term production broadly without specifying primary or secondary in their production dates - it is not necessarily synonymous for a filming date (primary production or original date of production).

We see this in the wild when looking at production dates from third-party retailers, or certain web sites, such as Karups. Karups makes a distinction between production date, filming date, and release date.

Users want to cite publisher production dates for the purpose of documenting original releases or calculating accurate performer scene ages, but given the broad usage of the term, this will often lead to inaccuracies.

### A better metadata field solution 
To maintain canonical accuracy and avoid interpretation errors I propose the following:

| Metadata field | Description|
|---|---|
| Filming date | The date the visual depiction was recorded on camera. (This would be the primary field used when calculating performer scene ages) |
| Production date | The exact 1:1 reflection of whatever the publisher claims as the production date. (Not to interpreted or adjusted since it could be primary or secondary production dates which would both be accurate)

# Original post

**Is your feature request related to a problem? Please describe.**
Many scenes, including website releases and movie scene compilations, are re-releases not accurately reflecting the original date of production.  This will reflect an inaccurate chronological representation of the scene, of the performers age during the scene, etc. 

**Describe the solution you'd like**
Similar to id3v2 and MusicBrainz tag metadata, add metadata fields for original release dates for scenes:
| Metadata field | Description |
| --- | --- |
| originaldate | Original date of production (YYYY-MM-DD) of the earliest release |
| originaldate_url | Source URL for the original date |

**Additional context**
~~i would consider these _advanced metadata_ fields that would not be commonly used. Hence, to minimize confusion or accidental usage, their input forms should not be displayed in the scene metadata editor by default. i propose displaying them when activated via the scene editor's hamburger menu > show advanced tags (which would just change the css display value from hidden to block, or something along these lines)~~

**Comment by AdultSun** [comment]
Same feature request but for Stash-Box: https://github.com/stashapp/stash-box/issues/346

**Comment by AdultSun** [comment]
Continuing the discussion from PR #6533 

---

As far as I understand, this is fundamentally a discussion of whether production dates (or any other secondary date field) is better handled as a core feature or with a custom field and plugin(s). Debates over how the current "Date" field is used, if it should be renamed, how we would handle migration, how closely Stash should mirror Stash-Box, etc. all seem secondary to me. We know people want production date support in Stash, so what's the best way to implement that? I still think it makes more sense as a core feature, but if the desired functionality can be handled some other way then I could be convinced.

Most of these have been mentioned already in the PR thread, but this is the functionality production dates could provide (let me know if I missed anything): 

1. **Submit to a Stash-Box** (draft submissions)
1. **Scrape from a Stash-Box** (StashDB, FansDB, TPDB, etc.)
1. **Scrape something that isn't a Stash-Box** (community scrapers)
1. **Basic sort / filter by production date**, including options exclusive to other date fields (might just be calendar button right now)
1. **Two-staged sort / filter by production date first, release date as fallback** (chronological sort accounting for null prod. dates)
1. **Two-staged sort / filter by release date first, production date as fallback** (same as default behavior but accounting for null release dates)
1. **Display performer age calculated with production date first, release date as fallback** (preferably default behavior, vice versa doesn't make much sense)
1. **Sort / filter by performer age calculated with production date first, release date as fallback** (preferably default behavior, vice versa doesn't make much sense)
1. **Sharing / syncing production dates between linked objects of varying types**, i.e. Groups <-> Scenes <-> Galleries <-> Images (definitely a complicated feature that wouldn't apply to all situations)
1. **Sharing / syncing production dates between linked versions of the same content**, i.e. Original <-> Remaster <-> DVD rip <-> Re-distribution (the often discussed "release groups" concept for Stash-Box)

How difficult would it be to implement all of these with a custom field / plugin compared to a core feature? I'm not sure everything in this list is even possible with a custom field / plugin right now, particularly anything that requires calculations and comparisons to other dates (release dates, birthdates). And when it comes to interactions with both Stash-Box and other scrapers, it seems to me like a custom production date field would still need to be added to the schema(?) of those systems. And at that point, I don't know what the difference would be between an officially sanctioned custom field and a hardcoded field, or why the former would be preferable to the latter.

And even if all of this is feasible outside of a core feature, how difficult would it be to discover, setup, and access this functionality? I don't think interest is limited only to power users, so any unnecessary friction could make it inaccessible to much of its intended audience.

As it stands, few editors in StashDB interact with production dates at all, but I suspect that's largely because they aren't tied into any functionality other than being saved and displayed on StashDB. I'm concerned that even if support is expanded in Stash, if it comes with too much friction then it could still blunt editors' interest in curating production dates. Or if interest grows anyway, it could even create a headache for a stash-box's mods asked to troubleshoot any pain points or confusion from those struggling with that friction. We already have to do the same for editors struggling with community scrapers, for example.

I realize that's a lot of speculation about secondary or even tertiary consequences but remember, as far as I know stash-boxes are the *only* scrapable source of production dates right now, and Stash would potentially be the only local software to use them. That means Stash's implementation (or lack thereof) will directly impact the demand for that data moving forward, and projects like StashDB will have to deal with the result, whatever that may be.

---

I guess my larger point is that simply deciding production dates can be easily *stored* as a custom field doesn't address how that custom field can be _used_. What are the technical limitations introduced by that decision? What could it look like for the average user? And how could it impact the maintainers of related projects, like Stash-Box devs and StashDB editors?

I genuinely don't know the answers to those questions, and apparently we've been talking about production dates for literally five and a half years (at least on Github).

**Comment by AdultSun** [comment]
@echo6ix Sorry for the late response. I did see your earlier discussion on Discord about this, but I was out of town at the time and it slipped through the cracks after I got back home.

> Since writing this, I've learned that my understanding of production date was not completely correct. Because the post hinges on that definition, I'm not sure my original conclusions below are as valid as written.

Honestly, I don't think any legal definition of "production date" is really that relevant to us. I don't even think preserving the legal "Date of Production" or "Date of Manufacture" (which I've also seen on some DVD back covers) as a distinct and separate field has much value to either local Stash collections or community databases like StashDB. Preserving those exact official dates really only matters to the content producers / distributors and I guess certain governmental authorities. For the rest of us, it's just another resource we can use to determine a "production date" that's actually useful for users of Stash or Stash-Box.

I think of it as similar to the "sort name" field we've added to tags. Sometimes the default value isn't specific or accurate enough for the various functions that hook into it, so overriding it with a secondary value would be useful. "Sort names" allow users to define a new sort order for tags without losing the clean display name. This would be the same, allowing users to define a new date for various functions (sort, filter, performer age, etc.) without losing the release date.

You mention "filming date" as an alternate name, but I think that term is too specific. Remember, we don't know what kind of content that field will be applied to. Animations, fan compilations, photos, illustrations, audio, text, games... Obviously not all of those are natively supported by Stash right now, but we should still consider how a "production date" concept can be extended past our initial assumption of original live-action video. Any other terms that are too broad like "original date", "alternate date", "override date", etc. wouldn't signal what assumptions other features within the software (i.e. performer ages) make about what that date represents. Another option is "creation date", but that term is almost identical to the "Created At" field describing the database entry.

I still think "production date" is the term that makes the most sense for us. It's broad enough to provide some flexibility without being so obscure that nobody will know what it's supposed to mean. I also don't think its similarity to a legal "Date of Production" is a problem. Very few users are going to look at "production date" and immediately leap to the exact legal definition of "produce" in 2257 documentation as the only possible meaning and assume the software follows that strict definition. And if they make the connection between "production date" in Stash and a "Date of Production" listed on a DVD cover or intro slate anyway, I still see that as a good thing. Those dates still closely align with our intended usage and are still more useful and available than alternative sources. And after all, any inaccuracies introduced by a legal "Date of Production" can't be any worse than the accuracy of a scraped "release date", especially when Stash relies on those for the exact same functions.

**Comment by echo6ix** [comment]
I'm going to reply from the perspective of someone that has an interest in the data-integrity on StashDB.

I understand synergy between these two projects doesn't need to be 1:1, but I think you always want to avoid design decisions that might negatively impact the other.

### TLDR answer
In the current state of both Stash and stash-box, "production dates" and other auxiliary dates are a huge clusterfuck and should be left as a custom field with no out-of-the-box mechanism that submits it back into a stash-box.

I think it would be better to work on making custom fields more robust and manageable - it would benefit both a production date and any other field not currently supported that people may want in the future.

### Long, nuanced answer

> 1.Submit to a Stash-Box 

 Bad idea

> 2. Scrape from a Stash-Box (StashDB, FansDB, TPDB, etc.)

Good idea

> 3. Scrape something that isn't a Stash-Box (community scrapers)

Good idea (possibly)

> 4. Basic sort / filter by production date, including options exclusive to other date fields (might just be calendar button right now)

Good idea?

> 5. Two-staged sort / filter by production date first, release date as fallback (chronological sort accounting for null prod. dates)
> 6. Two-staged sort / filter by release date first, production date as fallback (same as default behavior but accounting for null release dates)

Unclear if this creates design issues based on user expectations with whatever they're clicking to sort

> 7. Display performer age calculated with production date first, release date as fallback (preferably default behavior, vice versa doesn't make much sense)
> 8. Sort / filter by performer age calculated with production date first, release date as fallback (preferably default behavior, vice versa doesn't make much sense)

Bad idea.

Locally people can define *production date* however they want. But if they're fetching this metadata from official or third-party sources (where they're mostly available) then this would be a hard-coded feature built on a fundamentally flawed premise that `production date = filming date = I can generate accurate ages`.

I think you read the edited first post, but just for others reading the follow-up, per the 2257 (better definitions in the first post):
- Produce, Production = any primary production or secondary production
- Primary production = filming date of the visual depiction (aka original date of production)
- Secondary production = something like manufacture date and will include manufacture dates of reissued content (so you're not getting "original" dates on reissued stuff), etc.

When the consumer sees published "production date" on third party retailers or elsewhere, it could mean either primary or secondary production. The publisher or retailer doesn't need to specify, and we won't know without doing detective work (which mostly yields no results for the majority of content).

So the entire idea of generating accurate ages based on this field are going to give mixed if not mostly inaccurate results. Using a filming date (to represent primary production) would resolve that misunderstanding mostly, but obtaining that is both difficult and rare).

> 9. Sharing / syncing production dates between linked objects of varying types, i.e. Groups <-> Scenes <-> Galleries <-> Images (definitely a complicated feature that wouldn't apply to all situations)
> 10. Sharing / syncing production dates between linked versions of the same content, i.e. Original <-> Remaster <-> DVD rip <-> Re-distribution (the often discussed "release groups" concept for Stash-Box)

Just for the reader's clarity, an "original", "remaster", and "redistribution" would each typically have different production dates. It would of course depend on whether they're referring to primary or secondary production (which they do not clarify), but based on my experience with conventional content it's typically secondary production dates.

And this leads me to my new opinion on these fields...

### Custom fields + plugins for age calculations

If these auxiliary dates were metadata fields that Stash didn't have the ability to submit back into StashDB I would advocate for their inclusion.

Since the original post, five years is a lot of time to see how people submit and vet content on StashDB. Until stash-box has a way to mandate and verify proper references for specific fields, this could be a bad idea if data integrity is a concern given the following:

- Most people don't understand that the production dates they see aren't necessarily filming dates (until recently I was one of them)
- When they realize the production date isn't giving them the correct performer scene age calculation guess what they're going to edit? The production date field.
- Fudging data locally is a user's prerogative, but if integration with stash-box is built in... they will submit potentially bad data back into StashDB without an easy way for voters/future editors to verify something that's typically not available on basic studio links that accompany submissions.

Therefore, refining custom fields (from global management to better local integration) to be more robust is probably better time spent for dev resources. More here https://github.com/stashapp/stash/issues/6795  -- more to my example in that thread, custom fields could get global uuid's or something like that. These could be defined by the user, or created semi-automatically much like performer or tags that are not detected locally upon a scrape).

Edit: I also didn't notice your last reply. I guess I was writing _this_ as you were typing _that_ =)

**Comment by AdultSun** [comment]
> Locally people can define production date however they want. But if they're fetching this metadata from official or third-party sources (where they're mostly available) then this would be a hard-coded feature built on a fundamentally flawed premise that `production date = filming date = I can generate accurate ages`.

You can say the exact same thing about the current "Date" field:  

"_Locally people can define **"Date"** however they want **(though I think it's almost universally seen as "release date")** but... it's still a hard-coded feature built on a fundamentally flawed premise that `**date** = **release date** = I can generate accurate ages`._"

Obviously that last statement about calculating ages from release dates makes even less sense, but that's how it currently works in Stash. And sure, not every official "Date of Production" will be an accurate representation of the exact filming date, but how often will it be *worse* than using the release date as a stand-in? And how does that compare to how often a sourced release date isn't an accurate representation of the exact date that content was made available to the public?

Replacing some of those hardcoded assumptions with separate "Release Date" and "Production Date" fields would be a step forward in terms of clarity and accuracy, not backward.

> Since the original post, five years is a lot of time to see how people submit and vet content on StashDB. Until stash-box has a way to mandate and verify proper references for specific fields, this could be a bad idea if data integrity is a concern given the following:
> 
> - Most people don't understand that the production dates they see aren't necessarily filming dates (until recently I was one of them)
> - When they realize the production date isn't giving them the correct performer scene age calculation guess what they're going to edit? The production date field.
> - Fudging data locally is a user's prerogative, but if integration with stash-box is built in... they will submit potentially bad data back into StashDB without an easy way for voters/future editors to verify something that's typically not available on basic studio links that accompany submissions.

Again, you could say exactly the same thing about release dates:

- _Most people don't understand that the **release dates** they see aren't necessarily accurate **(EX: studio changes to newer date, physical release vs. digital release, etc.)**_
- _When they realize the **release date** isn't giving them the correct performer scene age calculation guess what they're going to edit? The **release date field**_
- _Fudging data locally is a user's prerogative, but if integration with stash-box is built in... they will submit potentially bad data back into StashDB without an easy way for voters/future editors to verify..._


I cut off the last part of your last point with an ellipsis because release dates *are* typically available from studio links. But even so, I'd argue from experience that it's *easier* to spot suspicious production dates than release dates. If I see a production date listed in a pending edit without any source in the edit comments or anywhere else obvious (sometimes it's spelled out in the scene description), that's an automatic downvote from me. If there's no way to tell where it came from or how accurate it might be, it should be removed.

Now compare that to release dates. How easy is it to look at the release date in a pending edit with the initial comment "scraped from studio" and know that you should be suspicious of its accuracy? And even if there's no source provided in the initial comment, how often does the average reviewer compare the release date to the studio link or anywhere else before assuming it's accurate?

Practically, I don't see anything about production dates that would make them any higher risk than what's already covered on StashDB. Any risks or issues they present are the same ones presented by everything else, and any possible changes to address those issues would affect production dates no more or less than the rest of the database.

**Comment by echo6ix** [comment]
> You can say the exact same thing about the current "Date" field [...]

Yes, you can, and I have.

> [...] but how often will it be worse than using the release date as a stand-in? [...]

See comment directly above this.

> And how does that compare to how often a sourced release date isn't an accurate representation of the exact date that content was made available to the public?

We're getting into how people choose to catalog their data and if I understand correctly what is an acceptable level of estimated metadata for StashDB. Of course we all have different standards here - some people have rigid data integrity standards because their goal is precise canonical cataloging - and others have less rigid standards because their goals are more pragmatic.

> Again, you could say exactly the same thing about release dates

The difference that counters those concerns I listed is while the user can put whatever data they want in the `date` field locally, it must conform to your definition of `release date` once submitted to StashDB.

After five plus years of this, the quality of submitted content that is not sourced by a community scraper and relies on manual user input and interpretation - such as a production date - has made me more cynical not less. So this basically boils down to my cynicism that a synchronizable `production date` field will get grossly misused and submitted without much oversight. 

### Last words

As you noted in your prior post. You referenced having a date field that could support a diverse amount of content, like animations.

If a big tent design is where @WithoutPants wants to take Stash, then I think that's a strong case not for hardcoding more fields, but focus on refining the custom fields. Make them more robust, easier to use and manage (like global native fields), and give them the ability to integrate with scrapers and stash-box.

That would check a lot of boxes on the production date wish list you cited. It might not provide everything, but some of those features seem highly specified if Stash is to move into a big tent direction.

Based on how custom fields get refined, doing an age calculus based on their values seems like it would be a simple plugin that anyone could vibe code.

**Comment by AdultSun** [comment]
> We're getting into how people choose to catalog their data and if I understand correctly what is an acceptable level of estimated metadata for StashDB. Of course we all have different standards here - some people have rigid data integrity standards because their goal is precise canonical cataloging - and others have less rigid standards because their goals are more pragmatic.

Which is why StashDB has guidelines, to align the database's standards with those of the community's as best we can. Of course, how close we are to accomplishing is up for debate. A crowdsourced database will always be a work-in-progress.

> The difference that counters those concerns I listed is while the user can put whatever data they want in the date field locally, it must conform to your definition of `release date` once submitted to StashDB.

No that's not different, that's the same. Submissions of production dates have to match StashDB's definition, just like titles, descriptions, cover images, tags, etc. It's no more or less risky to include production dates in a draft submission as it is to include any of those others.

If anything, the fact that there are few (if any) ways to scrape a production date makes it more likely that the field will be _ignored_ when creating a draft submission than result in bad data, especially since it's so easy to spot a bad production date with no sourcing in the comment.

> After five plus years of this, the quality of submitted content that is not sourced by a community scraper and relies on manual user input and interpretation - such as a production date - has made me more cynical not less. So this basically boils down to my cynicism that a synchronizable production date field will get grossly misused and submitted without much oversight.

See, this is my whole point. The way I see it, all of these arguments on behalf of StashDB don't actually have much to do with the pros and cons of handling production dates specifically through hardcoding vs. custom fields. What all of these arguments *actually* add up to is a desire to limit the amount of discretion afforded to individual editors and increasing the level of documentation and scrutiny applied to data saved and submitted to the database.

I'm not saying that's an unfair criticism, to be clear, but it's a completely different conversation than this one.

> That would check a lot of boxes on the production date wish list you cited.

That isn't my personal wishlist of features, it's a rundown of possible functions that could hook into a separate production date field. Early on in the PR comment thread linked above, there were questions about what purpose a production date could serve in Stash other than simple storage. These are my answers to that question, mostly drawn from later comments from that thread.

Personally, I'm only concerned about the expanded sorting / filtering options, more accurate age calculations, and Stash-Box integrations. The last two items in that list (syncing prod. dates across objects and within release groups) are more pie-in-the-sky suggestions to illustrate how prod. dates *could* be utilized in Stash and Stash-Box long-term, rather than actionable feature requests in the short-term.

> Based on how custom fields get refined, doing an age calculus based on their values seems like it would be a simple plugin that anyone could vibe code.

That first phrase — "based on how custom fields get refined" — does a lot of heavy lifting here. I don't know what is or isn't possible for custom fields, what intentions the maintainers might have for them, or at what point we could realistically expect the system to be mature enough that "anyone could vibe code" a plugin to handle any of the functions in my list, if at all. I also don't know enough to make any assumptions, which is why all I can do is ask for others to clarify.

---

Those are the same kinds of questions I was asking in my first post today. They're not rhetorical questions either. It's been five and half years since this issue was created, and in all that time I still do not understand how custom fields and plugins could be a sufficient solution compared to the hardcoded field I've always imagined.

Instead, the conversation lately has drifted to debates over how the "Date" field in Stash is used, what assumptions we have or should make about it, whether it should be renamed "Release Date" or not, how closely aligned Stash and Stash-Box functionality should be in general, what the standards should be for closing or re-opening PRs, the legal definition of "produce" in regards to 2257 documentation, and now the data integrity of StashDB writ large. Again, those are all topics worth discussing in their own time and place, but none of them address the core question: If a hardcoded production date isn't the best solution, then what is?

And yet, a relatively straightforward PR was just closed, seemingly in favor of a separate approach that to my untrained eye looks much more difficult and convoluted but hasn't received the same level of scrutiny. If that's *not* the case and using custom fields really is the more practical solution, then someone please explain how before another five and half years passes us by. I feel I've done more than my fair share of explaining already.
